rudder-rest/src/main/scala/com/normation/appconfig/ConfigService.scala

```diff
@@ -334,27 +354,55 @@ class LDAPBasedConfigService(configFile: Config, repos: ConfigRepository, workfl
     }
   }
 
-  private[this] def save[T](name: String, value: T, modifyGlobalPropertyInfo : Option[ModifyGlobalPropertyInfo] = None): Box[RudderWebProperty] = {
-    val p = RudderWebProperty(RudderWebPropertyName(name), value.toString, "")
+  private[this] def save[T](name: String, value: T, modifyGlobalPropertyInfo : Option[ModifyGlobalPropertyInfo] = None)(implicit ser: T => String): Box[RudderWebProperty] = {
+    val p = RudderWebProperty(RudderWebPropertyName(name), ser(value), "")
     repos.saveConfigParameter(p,modifyGlobalPropertyInfo)
   }
 
+  private[this] implicit def serInt(x: Int): String = x.toString
+  private[this] implicit def serPolicyMode(x: PolicyMode): String = x.name
+  private[this] implicit def serFeatureSwitch(x: FeatureSwitch): String = x.name
+
+
   private[this] implicit def toBoolean(p: RudderWebProperty): Boolean = p.value.toLowerCase match {
     case "true" | "1" => true
     case _ => false
   }
+  private[this] implicit def serBoolean(x: Boolean): String = if(x) "true" else "false"
 
   private[this] implicit def toOptionBoolean(p: RudderWebProperty): Option[Boolean] = p.value.toLowerCase match {
     case "true" | "1" => Some(true)
     case "none" => None
     case _ => Some(false)
   }
+  private[this] implicit def serOptionBoolean(x: Option[Boolean]): String = x match {
+    case None        => "none"
+    case Some(true)  => "true"
+    case Some(false) => "false"
+  }
 
   private[this] implicit def toSyslogProtocol(p: RudderWebProperty): SyslogProtocol = p.value match {
     case SyslogTCP.value => // value is TCP
       SyslogTCP
     case _ => SyslogUDP
   }
+
+  private[this] implicit def serSyslogProtocol(x: SyslogProtocol): String = x.value
```

Comment:
Why this one ? It's not a per node configuration

Reply:
It's needed because of the more strict requirement of serializer. Now, we need one for each type, where before "toString" (line 338 above) was used (but it's a bad default most of the time).